anabot/runtime/installation/hub/partitioning/__init__.py

```diff
@@ -115,6 +115,78 @@ def disk_handler(element, app_node, local_node):
 def disk_check(element, app_node, local_node):
     return disk_manipulate(element, app_node, local_node, True)
 
+# Anaconda doesn't provide enough information for ATK about disk selection
+# so we need to remember the disk selection ourself.
+__spec_disk_selection = {}
+def spec_disk_manipulate(element, app_node, local_node, dryrun):
```

Comment:
Hmm, I probably should have rephrased the comment and removed the BZ link (although I have no idea if this is still a problem on ppc64). However, the main point I didn't change the way the selection is checked is that I don't know how to do that properly. The only hint I could find is that the disk icon changes its text depending on whether it's selected or not (e. g. it says 'Hard Disk' for an unselected disk an '' for a selected one) - nevertheless I'm not really sure this is something we should rely on as I have no idea if this works under at all times or can't change under some circumstances. Therefore I wouldn't change a piece of code that has worked OK so far without being sure I won't break it by such a change. Moreover, I think it would need some more work & time that's not worth it.

Regarding the copy-pasting vs. code sharing, I wanted to use something that is proven to work without possibly breaking the existing code, but you're right it might deserve sharing the main code. Do you think it would thus make sense to rename `disk_manipulate` to e. g. `local_disk_manipulate`, move (most of) the existing code to `disk_manipulate` and extend it to take the disk type and disk selection dict as parameters? And also change `spec_disk_manipulate` accordingly to use the common `disk_manipulate` function?

Reply:
Ok, thanks, this explanation basically belongs to the comment that there doesn't seem to be other reliable method for this describing the only other available option that was available during the implementation. This comment makes it easier for the maintenance as it's easy to assess if such code is still available (as the comment is still valid) or if the code can be changed as there's another reliable method available (the comment is not valid anymore). The copy-pasted comments made me wondering if it's still true.

I don't have any preference on how to share the code, I'm ok with renaming the `disk_manipulate` function to some other name (I'd just avoid the word local as the device afaik does not have to be necessarily local) such as `basic_disk_manipulate`. The code can be done in a way that it consists of multiple simpler functions being called (may be easier to understand) or one bigger "spaghetti-like" parameterized code - it depends which solution will be easier to maintain. I have a feeling and hope that the one common code should not contain so many if statements. Thinking about the biggest if is about the pre-selected drives but I have no idea if there may be some drives already present in the advanced storage - maybe iSCSI or DASD.

Reply:
Ok, so there's another turn. Let's not implement this at all as @jstodola pointed out that there are 2 ways how the device can be deselected (in the partitioning spoke or in the advanced devices subspoke). Additionally, any device that is selected in the advanced devices subspoke should become automatically selected in the partitioning spoke which means that we effectively don't need to interact with the advanced storage devices in the partitioning spoke.

Ignoring this GUI part limits the possibilities of GUI testing but doesn't limit necessary interactions with the advanced storage devices.

Reply:
OK, I removed the code for specialized disks selection in the main partitioning spoke as we'd discussed.